### ISSUE TITLE
eqeqeq Rule Change Proposal: Allow only == undefined

### ISSUE THREAD

**Issue by kirly-af** [eslint/eslint#7658]
**What rule do you want to change?**

eqeqeq

**Does this change cause the rule to produce more or fewer warnings?**

Both fewer and more. See example below.

**How will the change be implemented? (New option, new default behavior, etc.)?**

It will be a new option. Currently we can do:
```
["error", "always", {"null": "ignore"}]
```

The new option would allow to set the rule this way:

```
["error", "always", {"undefined": "ignore"}]
```

Thus, that would allow the use of `==` and `!=` only in the case one compares to `undefined`.

**Please provide some example code that this change will affect:**

It will be beneficial to people who want to use `undefined` exclusively, instead of `null`, but keep the possibility to check both `null` and `undefined`.

Although people might forbid null in their code, it still has to be checked in case it is returned in some third party library call, or JSON deserialized objects.

``` js
const foo = someThirdPartyFunction()
const bar = JSON.parse(json)

if (foo == undefined && bar != undefined) {
  // ...
}

if (foo == null && bar != null) {
  // ...
}
```

**What does the rule currently do for this code?**

Currently the following code is **invalid** :
``` js
// ["error", "always", {"null": "ignore"}]

const foo = someThirdPartyFunction()
const bar = JSON.parse(json)

if (foo == undefined && bar != undefined) {
  // ...
}
```

Currently the following code is **valid** :
``` js
// ["error", "always", {"null": "ignore"}]

const foo = someThirdPartyFunction()
const bar = JSON.parse(json)

if (foo == null && bar != null) {
  // ...
}
```

**What will the rule do after it's changed?**
The following code would be **valid** :

``` js
// ["error", "always", {"undefined": "ignore"}]

const foo = someThirdPartyFunction()
const bar = JSON.parse(json)

if (foo == undefined && bar != undefined) {
  // ...
}
```

The following code would be **invalid** :
``` js
// ["error", "always", {"undefined": "ignore"}]

const foo = someThirdPartyFunction()
const bar = JSON.parse(json)

if (foo == null && bar != null) {
  // ...
}
```

**Comment by platinumazure** [comment]
Hi, could you please fill out our [rule enhancement request template](https://github.com/eslint/eslint/blob/master/templates/rule-change-proposal.md)?

**Comment by eslintbot** [comment]
Hi @kirly-af, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by vitorbal** [comment]
ping @kirly-af, could you look at @platinumazure's question above? Thanks!

**Comment by kirly-af** [comment]
Hi @vitorbal, @platinumazure 

Sorry I've been quite busy recently. I've updated the proposal, hopefully now there's enough information.

**Comment by platinumazure** [comment]
@kirly-af Thank you very much for updating the initial post, really appreciate it.

I'm :+1: on the idea, but we may need to bikeshed the configuration option a bit.

**Comment by kirly-af** [comment]
Any news on this one ?

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.

**Comment by Schweinepriester** [comment]
+1

---

> Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.

There hardly has been any feedback.
